<!-- jplaced@unizar.es
 2022, Universidad de Zaragoza -->

<!-- Move base with SLAM karto -->

<launch>
  <master auto="start"/>

  <arg name="namespace"/>
  <arg name="robot_x"/>
  <arg name="robot_y"/>
  <arg name="robot_z"/>
  <arg name="robot_R"/>
  <arg name="robot_P"/>
  <arg name="robot_Y"/>
  <param name="use_sim_time" value="true" />

  <node pkg="slam_karto_g2o" type="slam_karto_g2o" name="slam_karto_g2o" output="screen">
    <rosparam command="load" file="$(find graph_d_exploration)/param/$(arg namespace)/mapping_karto_g2o.yaml" />
    <param name="base_frame" value="$(arg namespace)/base_link"/>
    <param name="odom_frame" value="$(arg namespace)/odom"/>
    <param name="map_frame" value="map"/>
    <param name="namespace" value="$(arg namespace)"/>
  </node>

  <node pkg="move_base" type="move_base" respawn="false" name="move_base_node" output="screen">
    <!-- Time elastic band local planner -->
    <!-- <param name="base_global_planner" value="navfn/NavfnROS"/> -->
    <!-- <param name="base_local_planner" value="dwa_local_planner/DWAPlannerROS"/> -->
    <param name="controller_frequency" value="50.0" />

    <rosparam file="$(find graph_d_exploration)/param/$(arg namespace)/costmap_common_params.yaml" command="load" ns="global_costmap" />
    <rosparam file="$(find graph_d_exploration)/param/$(arg namespace)/costmap_common_params.yaml" command="load" ns="local_costmap" />
    <rosparam file="$(find graph_d_exploration)/param/local_costmap_params.yaml" command="load" />
    <rosparam file="$(find graph_d_exploration)/param/global_costmap_params.yaml" command="load" />
    <rosparam file="$(find graph_d_exploration)/param/trajectory_planner.yaml" command="load" />
    <rosparam file="$(find graph_d_exploration)/param/base_local_planner_params.yaml" command="load" />
    <rosparam file="$(find graph_d_exploration)/param/base_global_planner_params.yaml" command="load" />

    <param name="max_planning_retries" value="-1"/>
    <param name="recovery_behavior_enabled" value="true"/>
    <rosparam file="$(find graph_d_exploration)/param/recovery_behaviors.yaml" command="load" />

    <remap from="/move_base_simple/goal" to="move_base/goal"/>
  </node>

</launch>
